index.bs

```diff
@@ -5474,14 +5494,213 @@ A [=remote end=] has a <dfn>navigable cache behavior map</dfn> which is a weak
+an [=struct/item=] named <code>cloned body</code>, which is a [=response/body=] or null,
+an [=struct/item=] named <code>collectors</code>, which is a list of <code>network.Collector</code>,
+an [=struct/item=] named <code>request</code>, which is a <code>network.Request</code>,
+an [=struct/item=] named <code>size</code>, which is a js-uint or null,
+an [=struct/item=] named <code>type</code>, which is a <code>network.BodyType</code>.
+
+A <dfn for="network">collector</dfn> is a [=/struct=] consisting of
+an [=struct/item=] named <code>body max size</code>, which is a js-uint,
+an [=struct/item=] named <code>contexts</code>, which is a list of <code>browsingContext.BrowsingContext</code>,
+an [=struct/item=] named <code>body types</code>, which is a list of <code>network.BodyType</code>,
+an [=struct/item=] named <code>collector</code>, which is a <code>network.Collector</code>,
+an [=struct/item=] named <code>collector type</code>, which is a <code>network.CollectorType</code>,
+an [=struct/item=] named <code>user contexts</code>, which is a list of <code>browser.UserContext</code>.
+
+A [=BiDi session=] has <dfn>network collectors</dfn> which is a [=/map=] between
+<code>network.Collector</code> and a [=network/collector=]. It is initially empty.
+
+A [=remote end=] has <dfn>collected network bodies</dfn> which is a list of
+[=network/body|bodies=]. It is initially empty.
+
+A [=remote end=] has a <dfn>max total body size</dfn> which is an integer representing
+the size allocated to collect network bodies in [=collected network bodies=]. Its
+value is implementation-defined.
+
+<div algorithm>
+To <dfn>get navigable for request</dfn> given request:
+
+1. Let |navigable| be null.
+
+1. If |request|'s [=request/window=] is an [=environment settings object=]:
+
+   1. Let |environment settings| be |request|'s [=request/window=]
+
+   1. If there is a [=/navigable=] whose [=active window=] is |environment
+      settings|' [=environment settings object/global object=], set |navigable|
+      to be that navigable.
+
+1. Return |navigable|.
+
+</div>
+
+<div algorithm>
+To <dfn>match collector for navigable</dfn> given |collector| and |navigable|:
+
+1. If |collector|'s <code>contexts</code> is not [=list/empty=]:
+
+  1. If |collector|'s <code>contexts</code> [=list/contains=] |navigable|'s [=navigable id=], return true.
+
+  1. Otherwise, return false.
+
+1. If |collector|'s <code>user contexts</code> is not [=list/empty=]:
+
+  1. Let |user context| be |navigable|'s [=associated user context=].
+
+  1. If |collector|'s <code>user contexts</code> [=list/contains=] |user context|'s [=user context id=], return true.
+
+  1. Otherwise, return false.
+
+1. Return true.
+
+</div>
+
+<div algorithm>
+To <dfn export>clone network response body</dfn> given |request| and |response body|:
+
+Note: This hook is intended to be triggered by the fetch spec when the response is set.
+
+1. For each |session| in [=active BiDi sessions=]:
```

Comment:
this should not be per |session| since [=collected network bodies=] is per remote end (which I think it should be)?